### PR TITLE
Created basic xUnit tests

### PR DESCRIPTION
**Pull request by Justin-Lloyd**
I noticed there seemed to be no progress on this issue so I created a few xUnit tests, hope that's ok. 
I'll look to create some more over the next few days.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (Marusyk) [inline .vscode/launch.json]: Please don't commit this file

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (Marusyk) [inline .vscode/tasks.json]: Please don't commit this file

File: src/Grok.Net.Tests/UnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            GrokResult result = act.Parse("");
+
+            Assert.NotNull(result);
+            Assert.Empty(result);
+        }
+
+        [Fact]
+        public void PatternCountTest()
+        {
+            Grok act = new Grok("%{MONTHDAY:month}-%{MONTHDAY:day}-%{MONTHDAY:year} %{TIME:timestamp};%{WORD:id};%{LOGLEVEL:loglevel};%{WORD:func};%{GREEDYDATA:msg}");
```
Comment (Marusyk) [inline src/Grok.Net.Tests/UnitTests.cs]: Please follow AAA pattern

File: src/Grok.Net.Tests/UnitTests.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            Grok act = new Grok("");
+            GrokResult result = act.Parse("");
+
+            Assert.NotNull(result);
+            Assert.Empty(result);
+        }
+
+        [Fact]
+        public void PatternCountTest()
```
Comment (Marusyk) [inline src/Grok.Net.Tests/UnitTests.cs]: Could you explain the meaning of this test?

File: src/Grok.Net.Tests/UnitTests.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            Grok act = new Grok("%{MONTHDAY:month}-%{MONTHDAY:day}-%{MONTHDAY:year} %{TIME:timestamp};%{WORD:id};%{LOGLEVEL:loglevel};%{WORD:func};%{GREEDYDATA:msg}");
+            string logs = @"06-21-19 21:00:13:589241;15;INFO;main;DECODED: 775233900043 DECODED BY: 18500738 DISTANCE: 1.5165
+               06-21-19 21:00:13:589265;156;WARN;main;DECODED: 775233900043 EMPTY DISTANCE: --------";
+            GrokResult grokResult = act.Parse(logs);
+
+            Assert.Equal("21:00:13:589241", grokResult[3].Value);
+            Assert.Equal("21:00:13:589265", grokResult[11].Value);
+        }
+
```
Comment (Marusyk) [inline src/Grok.Net.Tests/UnitTests.cs]: Please remove am empty lines

File: .vscode/launch.json (hunk 1 above)
Comment (Justin-Lloyd) [reply .vscode/launch.json]: Done

File: .vscode/tasks.json (hunk 2 above)
Comment (Justin-Lloyd) [reply .vscode/tasks.json]: Done

File: src/Grok.Net.Tests/UnitTests.cs (hunk 3 above)
Comment (Justin-Lloyd) [reply src/Grok.Net.Tests/UnitTests.cs]: Done

File: src/Grok.Net.Tests/UnitTests.cs (hunk 5 above)
Comment (Justin-Lloyd) [reply src/Grok.Net.Tests/UnitTests.cs]: Done, with the exception that certain line breaks are retained to delineate between AAA sections (I'm not a big fan of typing //Arrange etc for each section in each test)

File: src/Grok.Net.Tests/UnitTests.cs (hunk 4 above)
Comment (Justin-Lloyd) [reply src/Grok.Net.Tests/UnitTests.cs]: The Grok pattern contains 8 elements, and the logs contains 2 entries. This test simply validates that the logs are correctly parsed, using the defined pattern, by testing that a total of 16 elements are returned.

**Comment by Justin-Lloyd** [comment]
> Thank you for your contribution. We need more tests that cover more cases. I think it should be at least 10 test that cover really meaningful cases

Happy to get involved!

**Comment by Marusyk** [comment]
@Justin-Lloyd your changes have been merged, thanks for your contribution 👍

**Pull request merged**